### DISCUSSION TITLE
Osquery not showing MacOS client

### DISCUSSION THREAD

**Discussion by kgoode517** [Security-Onion-Solutions/securityonion#7424] [Unsupported Versions] [answered]
I have attempted to follow along with getting Osquery set up for my Imac Monterey with both the directions from https://docs.securityonion.net/en/2.3/osquery.html which I followed exactly but the client never showed up on fleet. I then tried alsos doing a fleet-ctl build of the pkg according to https://blog.fleetdm.com/a-quick-guide-to-installing-osquery-and-enrolling-macos-devices-into-fleet-f11b00f68431 which installed the package but I have no idea where it put it. I can't seem to find the files on the system either. The client never appears on the fleet portal. How should I proceed with troubleshooting?

**Comment by defensivedepth** [comment]
Did you run `so-allow` on the Manager and configure osquery?

**Comment by kgoode517** [comment]
Josh, Thank you for taking a look. Yes I ran so-allow and have already added in the client for the osquery endpoint. I configured osquery according to the security onion documentation for macOS

**Comment by defensivedepth** [comment]
Ok, please restart osquery on the mac and post the startup logs.

`sudo launchctl kickstart -k system/com.so-launcher.launcher`

**Reply by kgoode517** [reply]
I have already attempted that and it did not add the host.

**Comment by kgoode517** [comment, marked as answer]
I was able to solve this issue by looking at the logs and reveiwing my working osquery instance on my windows machine. The hostname has to be the IP address not the actual hostname I wished this was clarified in the docs. I had missed the part of the documentation that stated how to review the logs by running sudo tail /var/log/so-launcher/launcher-stderr.log. I hope this helps the next person who gets stuck on this.

**Reply by cyb3rz3us** [reply]
**_"hostname has to be the IP address not the actual hostname"_** - This is not my experience.  I use the hostname all the time on the Macs I manage but the hostname must be populated in at least one of **(a)** whatever DNS you are using; **(b)** your ```/etc/hosts``` file.

**Reply by defensivedepth** [reply]
Yes, hostname or IP should work. As @cyb3rz3us pointed out, your local system will need to be able to resolve the hostname to the Manager IP.

**Reply by kgoode517** [reply]
That would make sense I guess I should have thought to add it to the /etc/hosts. I'm just glad it's working. Is there a reason the software inventory is disabled by default? That seems like a super useful feature. Is there a way to enable this easily?

**Reply by defensivedepth** [reply]
@kgoode517 RE: Disabling the vuln proc feature - see here: https://github.com/Security-Onion-Solutions/securityonion/discussions/7643#discussioncomment-2440468

**Reply by kgoode517** [reply]
Excellent,
That completely makes sense.
